src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -411,7 +411,7 @@ private void OnEndStreamReceived()
             }
 
             OnTrailersComplete();
-            RequestBodyPipe.Writer.Complete();
+            await RequestBodyPipe.Writer.CompleteAsync();
```

Comment:
When the call to CompleteAsync() is the only thing awaited in the method, try to just return the ValueTask directly rather than await it.